frontend/plugins/acm/resource-routes.ts

```diff
@@ -0,0 +1,63 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { generatePath } from 'react-router-dom-v5-compat'
+import queryString from 'query-string'
+import { ResourceRouteHandler } from '@stolostron/multicluster-sdk'
+
+/**
+ * ACM resource route handler for first-class resource pages.
+ * Routes resources to their dedicated ACM pages instead of generic search results.
+ */
+export const acmResourceRouteHandler: ResourceRouteHandler = ({ cluster, namespace, name, resource, model }) => {
+  const { kind, group } = model
+
+  switch (kind.toLowerCase()) {
+    case 'cluster':
+    case 'managedcluster':
+      return generatePath('/multicloud/infrastructure/clusters/details/:namespace/:name/overview', {
+        namespace: name,
+        name,
+      })
+
+    case 'application': {
+      if (group === 'app.k8s.io' || group === 'argoproj.io') {
+        const params = queryString.stringify({
+          apiVersion: `${kind}.${group}`.toLowerCase(),
+          cluster: resource._hubClusterResource ? undefined : cluster,
```

Comment:
_hubClusterResource is not an actual property that appears on k8s resources - it is something search adds. This TS error is being caught in VSCode and it should fail the builds, but I think we are not checking source files under frontend/plugin. So when you move it under frontend/src, this should show up as an error.